agent/config/builder.go

```diff
@@ -1410,6 +1410,12 @@ func (b *builder) validate(rt RuntimeConfig) error {
 		if err := s.Validate(); err != nil {
 			return fmt.Errorf("service %q: %s", s.Name, err)
 		}
+		// Check for errors in the check definitions
+		for _, c := range s.Checks {
+			if err := c.Validate(); err != nil {
+				return fmt.Errorf("check %q: %s", c.Name, err)
+			}
+		}
```

Comment:
Nice! This looks good for validating the checks in services. There can also be checks specified at the top level of the config (outside of services), and I believe we aren't validating those yet.

Unless this is happening somewhere else that I missed, I think we may need something like this:

```go
	for _, check := range rt.Checks {
		if err := check.CheckType().Validate(); err != nil {
			return fmt.Errorf("check %q: %w", check.Name, err)
		}
	}
```

Reply:
And indeed there are:
https://github.com/hashicorp/consul/blob/242b3a2dc574556e5dc34be1f679ac5f52463bf8/agent/config/runtime.go#L494
And I see you also added a call to `CheckType()`:
https://github.com/hashicorp/consul/blob/242b3a2dc574556e5dc34be1f679ac5f52463bf8/agent/structs/check_definition.go#L170-L173
I see the intention is to convert `CheckDefinition` into `CheckType`. Good call.